#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_invalid_indexes.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## DP: Fixes a crash, see
## DP: http://www.cae.wisc.edu/pipermail/bug-octave/2006-November/001077.html

@DPATCH@
diff -urNad octave2.9-2.9.9~/liboctave/Array.cc octave2.9-2.9.9/liboctave/Array.cc
--- octave2.9-2.9.9~/liboctave/Array.cc	2006-09-15 22:29:18.000000000 +0200
+++ octave2.9-2.9.9/liboctave/Array.cc	2006-11-07 08:30:44.000000000 +0100
@@ -3113,6 +3113,16 @@
 
 	      frozen_len = freeze (idx, new_dims, true);
 
+	      for (int i = 0; i < idx.length (); i++)
+		{
+		  if (! idx(i))
+		    {
+		      retval = 0;
+		      lhs.clear_index ();
+		      return retval;
+		    }
+		}
+
 	      if (rhs_is_scalar)
 		{
 		  if (n_idx < orig_lhs_dims_len)
